File: U3DExporter/2022_10_29_09_58_39_Android/Assets/InsightARWorld/InsightARExporter/Utility/BuiltinJSScripts/Landmark/LandmarkManager.js
var g_FwUtility;
var g_ResManager;
var g_StateCtrl;
var g_LandmarkManager;

function LandmarkManager(gameObject)
{
    this.transform = gameObject.transform;
    this.gameObject = gameObject;
}
//Write prototype function here
LandmarkManager.prototype = Object.assign(Object.create(Object.prototype), {
    // Start is called before the first frame update
    Start: function(){
        Insight.Debug.Log("landmarkmanager start");
        g_FwUtility = this.gameObject.getComponent("Assets/InsightARWorld/InsightARExporter/Utility/BuiltinJSScripts/Util/Fw_Utility.js",0);
        g_ResManager = this.gameObject.getComponent("Assets/InsightARWorld/InsightARExporter/Utility/BuiltinJSScripts/Landmark/LandmarkResource.js",0);
        g_StateCtrl = this.gameObject.getComponent("Assets/InsightARWorld/InsightARExporter/Utility/BuiltinJSScripts/Landmark/LandmarkStateCtrl.js",0);
        g_LandmarkManager = this.gameObject.getComponent("Assets/InsightARWorld/InsightARExporter/Utility/BuiltinJSScripts/Landmark/LandmarkManager.js",0);
        this.Init();
    },
    Awake: function(){

    },
    // Update is called once per frame
    Update: function()
    {
        
    },
    Init: function()
    {
        g_ResManager.Init();
        g_StateCtrl.Init();
    },
    OnDisable: function()
    {
	    //this.Close();
    },
    //关闭
    Close: function()
    {
        //g_StateCtrl:ChangeState(g_GameStatus.GAME_STATUS_IDLE);
        if (g_StateCtrl)
            g_StateCtrl.Close();
    }

});

//Return the script module
LandmarkManager

File: Assets/InsightARWorld/InsightARExporter/Utility/BuiltinLuaScripts/Landmark/LandmarkManager.lua


local LandmarkManager = {}


function LandmarkManager:New(game_object)
    if self ~= LandmarkManager then
        return nil, "LandmarkManager:New is nil"
    end
    local newinstance = setmetatable({}, {__metatable = {}, __index = LandmarkManager})
    newinstance.transform = game_object.transform
    transform = game_object.transform
    newinstance.gameObject = game_object
    gameObject = game_object
    return newinstance
end

function LandmarkManager:Awake()

	
end

function LandmarkManager:Start()
	Insight.Debug.Log("landmarkmanager start");
	g_FwUtility = self.gameObject:GetComponent("Assets/InsightARWorld/InsightARExporter/Utility/BuiltinLuaScripts/Util/Fw_Utility.lua",0);
	g_ResManager = self.gameObject:GetComponent("Assets/InsightARWorld/InsightARExporter/Utility/BuiltinLuaScripts/Landmark/LandmarkResource.lua",0);
	g_StateCtrl = self.gameObject:GetComponent("Assets/InsightARWorld/InsightARExporter/Utility/BuiltinLuaScripts/Landmark/LandmarkStateCtrl.lua",0);
	g_LandmarkManager = self.gameObject:GetComponent("Assets/InsightARWorld/InsightARExporter/Utility/BuiltinLuaScripts/Landmark/LandmarkManager.lua",0);
	self:Init();
end

function LandmarkManager:Init()
	g_ResManager:Init();
	g_StateCtrl:Init();
end

function LandmarkManager:OnDisable()
	self:Close();
end

--关闭
function LandmarkManager:Close()
	--g_StateCtrl:ChangeState(g_GameStatus.GAME_STATUS_IDLE);
end

return LandmarkManager
